src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -429,11 +429,7 @@ public int compareOffsets(final NakadiCursor first, final NakadiCursor second) t
 
     @Override
     public void validateCommitCursor(final NakadiCursor position) throws InvalidCursorException {
```

Comment:
Could we fix `position`? 
You call validateCursorForNulls, but after you call KafkaCursor.fromNakadiCursor, which basically does the same checks. From other side it is not clear why you call KafkaCursor.fromNakadiCursor and do not work with the KafkaCursor.